integration-cli/docker_cli_attach_unix_test.go

```diff
@@ -82,12 +82,7 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
 	cpty.Write([]byte{17})
 
 	select {
-	case err := <-errChan:
-		if err != nil {
-			buff := make([]byte, 200)
-			tty.Read(buff)
-			c.Fatalf("%s: %s", err, buff)
-		}
+	case <-cmdExit:
```

Comment:
Should this check for an error?

Reply:
Probably, yes. The detach shouldn't fail.

Reply:
It has an error `exit status 125`.

Reply:
I don't understand. I think we meant, should this be something like:
```go
case err := <-cmdExit:
  if err != nil {
   c.Fatalf("failed to start container: %v", err)
 }
```

Reply:
Yes, that is what I understood. My response was "no". That is what the old code did, and it fails. `cmdExit` contains an error with the value `exit status 125`.  I haven't had a chance to look into that yet, but it seems that sending a detach now causes a non-zero exit. That doesn't necessarily seem wrong to me. If you ctrl-Z a program it will exit with non-zero, so why wouldn't the detach?

Reply:
Oh, I see. didn't realize our detach was doing so. Thanks for the clarification.

Reply:
Makes sense.

Reply:
>  haven't had a chance to look into that yet, but it seems that sending a detach now causes a non-zero exit. 

If we changed the exit code for detach in cli then it is a breaking change. We shouldn't do that unless it is absolutely needed(I definitely think error code 125 is wrong for a clean detach) and if it is, it needs to be on top of the changelog for the next releases. We shouldn't  cover up our tests for cli breakage.

Reply:
Considering `ctrl-z` exits with 148, what would you expect the exit code for `docker run` to be when it was detached?

Reply:
I guess it somewhat makes sense to use the same, but I'd rather not break people if there is no clear use case that needs this change.

Reply:
I think the change comes from https://github.com/docker/cli/pull/755, although supposedly this test was only failing with the changes in https://github.com/docker/cli/pull/696.

It seems very unlikely that anyone would use the detach keys for a script. If you wanted to detach in a script you would use `docker run -d`.